frontend/src/components/DrawComponent.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Draw from 'ol/interaction/Draw.js';
+import { unByKey } from 'ol/Observable.js';
+import VectorLayer from "ol/layer/Vector";
+import VectorSource from "ol/source/Vector";
+
+const DrawComponent = ({ map, geometryType, onDeleteGeometries }) => {
+  const drawInteraction = useRef(null);
+  const [drawing, setDrawing] = useState(false);
+
+  useEffect(() => {
+    if (!map || !geometryType) return;
+
+    const drawLayer = new VectorLayer({
+      name: 'drawLayer',
+      source: new VectorSource()
+    })
+
+    drawInteraction.current = new Draw({
+      source: drawLayer.getSource(),
+      type: geometryType,
+    });
+
+    map.addInteraction(drawInteraction.current);
+
+    const drawStartKey = drawInteraction.current.on('drawstart', () => {
+      setDrawing(true);
+      changeCursor();
+    });
+
+    const drawEndKey = drawInteraction.current.on('drawend', () => {
+      setDrawing(false);
+      changeCursor();
+      map.addLayer(drawLayer);
+    });
+
+    return () => {
```

Comment:
It is indeed a good practice to use the return statement of a useEffect for cleanups!